Tools/Scripts/libraries/webkitcorepy/webkitcorepy/task_pool.py

```diff
@@ -296,6 +321,9 @@ def main(cls, name, loglevel, setup, setupargs, setupkwargs, queue, teardown, te
 
         cls.name = name
         cls.working = True
+        cls.name = name
+        cls.working = True
+        cls.infinite_task_queue = []
```

Comment:
This is a big change to `TaskPool`. Which is necessary, I think, to achieve the goal. However, we need to write `TaskPool` unit tests for this, and probably break the `TaskPool` changes into a different PR, since they're substantial.